src/frontend-samples/explode-sample/ExplodeTile.ts

```diff
@@ -0,0 +1,629 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { assert, BeEvent, BeTimePoint, ByteStream, compareStrings, Id64, partitionArray } from "@bentley/bentleyjs-core";
+import { Point3d, Range3d, Transform, Vector3d, XYZProps } from "@bentley/geometry-core";
+import { BatchType, ElementGraphicsRequestProps, FeatureAppearance, FeatureAppearanceProvider, FeatureAppearanceSource, GeometryClass, IModelTileRpcInterface, Placement3d, TileFormat, TileVersionInfo, ViewFlagOverrides } from "@bentley/imodeljs-common";
+import { GraphicBranch, ImdlReader, IModelApp, IModelConnection, RenderSystem, SceneContext, Tile, TileContent, TileDrawArgParams, TileDrawArgs, TileLoadPriority, TileRequest, TileTree, TileTreeOwner, TileTreeReference, TileTreeSupplier, Viewport } from "@bentley/imodeljs-frontend";
+import ExplodeApp from "./ExplodeApp";
+
+/** Data describing an element for the exploded effect. */
+interface ElementData {
+  elementId: string;
+  origin: Point3d;
+
+  boundingBox: Range3d;
+  transformWorld: Transform;
+}
+
+/** Information returned by the query for the elements to the backend. */
+interface ElementDataProps {
+  id: string;
+  origin: XYZProps;
+  bBoxHigh: XYZProps;
+  bBoxLow: XYZProps;
+  pitch: number;
+  roll: number;
+  yaw: number;
+}
+
+/** Parameters used to create a ElementTile. */
+interface TileParams {
+  centerOfMass: Point3d;
+  versionInfo: TileVersionInfo;
+  data: ElementData;
+}
+
+/** Identifies our TileTree for the supplier. */
+interface ExplodeTreeId {
+  name: string;
+  ids: string[];
+}
+
+/** Parameters for creating a Tile Tree */
+interface ExplodeTreeParams {
+  objectName: string;
+  data: ElementData[];
+  iModel: IModelConnection;
+  tileVersionInfo: TileVersionInfo;
+}
+
+/** Parameters for creating the overloaded draw argument. */
+interface ExplodeTileDrawArgsParams extends TileDrawArgParams {
+  explodeFactor: number;
+}
+
+/** Overloaded TileDrawArgs that also contain the current Explode Scaling Factor. */
+class ExplodeTileDrawArgs extends TileDrawArgs {
+  public explodeFactor: number;
+  constructor(params: ExplodeTileDrawArgsParams) {
+    super(params);
+    this.explodeFactor = params.explodeFactor;
+  }
+}
+
+/** Creates a transform matrix translating the range away from the point scaled. */
+function calculateExplodeTransform(centerOfElement: Point3d, displaceOrigin: Point3d, explodeScaling: number) {
+  const vector = Vector3d.createFrom(displaceOrigin);
+  vector.subtractInPlace(centerOfElement);
+  vector.scaleInPlace(explodeScaling - 1);
+  return Transform.createTranslation(vector).inverse()!;
```

Comment:
Add a comment here explaining that `inverse` will not returned `undefined` because the transform consists solely of translation therefore always invertible.